webui/react/src/pages/Cluster/ClusterHistoricalUsage.tsx

```diff
@@ -0,0 +1,218 @@
+import { Button, Col, Row } from 'antd';
+import dayjs from 'dayjs';
+import React, { useEffect, useState } from 'react';
+
+import Section from 'components/Section';
+import { parseUrl } from 'routes/utils';
+import { getResourceAllocationAggregated } from 'services/api';
+
+import css from './ClusterHistoricalUsage.module.scss';
+import ClusterHistoricalUsageChart from './ClusterHistoricalUsageChart';
+import ClusterHistoricalUsageCsvModal from './ClusterHistoricalUsageCsvModal';
+import ClusterHistoricalUsageFilters, {
+  ClusterHistoricalUsageFiltersInterface,
+} from './ClusterHistoricalUsageFilters';
+import { mapResourceAllocationApiToChartSeries, ResourceAllocationChartSeries } from './utils';
+
+export const DEFAULT_RANGE_DAY = 14;
+export const DEFAULT_RANGE_MONTH = 6;
+export const MAX_RANGE_DAY = 31;
+export const MAX_RANGE_MONTH = 36;
+
+export enum GroupBy {
+  Day = 'day',
+  Month = 'month',
+}
+
+const ClusterHistoricalUsage: React.FC = () => {
+  const [ chartSeries, setChartSeries ] = useState<ResourceAllocationChartSeries|null>(null);
+  const [ filters, setFilters ] = useState<ClusterHistoricalUsageFiltersInterface>({
+    afterDate: dayjs().subtract(1 + DEFAULT_RANGE_DAY, 'day'),
+    beforeDate: dayjs().subtract(1, 'day'),
+    groupBy: GroupBy.Day,
+  });
+  const [ isCsvModalVisible, setIsCsvModalVisible ] = useState<boolean>(false);
+  const [ isUrlParsed, setIsUrlParsed ] = useState<boolean>(false);
+
+  /*
+  * When filters changes update the page URL.
+  */
+  useEffect(() => {
+    if (!isUrlParsed) return;
+
+    const dateFormat = 'YYYY-MM' + (filters.groupBy === GroupBy.Day ? '-DD' : '');
+    const searchParams = new URLSearchParams;
+    const url = parseUrl(window.location.href);
+
+    // after
+    searchParams.append('after', filters.afterDate.format(dateFormat));
+
+    // before
+    searchParams.append('before', filters.beforeDate.format(dateFormat));
+
+    // group-by
+    searchParams.append('group-by', filters.groupBy);
+
+    window.history.pushState(
+      {},
+      '',
+      url.origin + url.pathname + '?' + searchParams.toString(),
+    );
```

Comment:
**nit:** use the `useHistory` hook from `react-router-dom` instead of the native `window.history`. Example:
```
const history = useHistory();
history.push(...);
```
The hook provides a wrapper for the native version, so when we reference the wrapped version we might miss out on routes that are pushed directly into the native version.